node/external/blockAPI/internalBlock.go

```diff
@@ -168,6 +170,102 @@ func (ibp *internalBlockProcessor) GetInternalStartOfEpochMetaBlock(format commo
 	return ibp.convertMetaBlockBytesByOutputFormat(format, blockBytes)
 }
 
+// GetInternalStartOfEpochValidatorsInfo wil return the epoch start validators info for the provided epoch
+func (ibp *internalBlockProcessor) GetInternalStartOfEpochValidatorsInfo(epoch uint32) ([]*state.ShardValidatorInfo, error) {
+	if ibp.selfShardID != core.MetachainShardId {
+		return nil, ErrMetachainOnlyEndpoint
+	}
+
+	storer, err := ibp.store.GetStorer(dataRetriever.MetaBlockUnit)
+	if err != nil {
+		return nil, err
+	}
+
+	epochStartIdentifier := core.EpochStartIdentifier(epoch)
+	blockBytes, err := storer.GetFromEpoch([]byte(epochStartIdentifier), epoch)
+	if err != nil {
+		return nil, err
+	}
+
+	metaBlock := &block.MetaBlock{}
+	err = ibp.marshalizer.Unmarshal(metaBlock, blockBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	validatorsInfo, err := ibp.getAllValidatorsInfo(metaBlock)
+	if err != nil {
+		return nil, err
+	}
+
+	return validatorsInfo, nil
+}
+
+func (ibp *internalBlockProcessor) getAllValidatorsInfo(metaBlock data.HeaderHandler) ([]*state.ShardValidatorInfo, error) {
+	allValidatorInfo := make([]*state.ShardValidatorInfo, 0)
+	for _, miniBlockHeader := range metaBlock.GetMiniBlockHeaderHandlers() {
+		hash := miniBlockHeader.GetHash()
+
+		miniBlock, err := ibp.getMiniBlockByHash(hash, metaBlock.GetEpoch())
+		if err != nil {
+			return nil, err
+		}
+
+		if miniBlock.Type != block.PeerBlock {
+			continue
+		}
+
+		validatorInfo, err := ibp.getValidatorsInfo(miniBlock)
+		if err != nil {
+			return nil, err
+		}
+
+		allValidatorInfo = append(allValidatorInfo, validatorInfo...)
+	}
+
+	return allValidatorInfo, nil
+}
+
+func (ibp *internalBlockProcessor) getValidatorsInfo(miniBlock *block.MiniBlock) ([]*state.ShardValidatorInfo, error) {
```

Comment:
this works only with the peer-miniblocks feature enable. Do we want it work also with the pre-peer-miniblocks feature types of miniblocks?

Reply:
added handling for endpoint before peer refactor activation